// @react-login-page/page10/cjs/control/login/Email.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Email = void 0;
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = require("react");
var _reactLoginPage = require("react-login-page");
var _email = require("../../icons/email");
var _jsxRuntime = require("react/jsx-runtime");
var _excluded = ["keyname", "name", "visible", "rename", "label", "panel"];
var Email = exports.Email = /*#__PURE__*/(0, _react.memo)(function (props) {
  var _props$keyname = props.keyname,
    keyname = _props$keyname === void 0 ? 'email' : _props$keyname,
    name = props.name,
    _props$visible = props.visible,
    visible = _props$visible === void 0 ? true : _props$visible,
    rename = props.rename,
    label = props.label,
    _props$panel = props.panel,
    panel = _props$panel === void 0 ? 'login' : _props$panel,
    elmProps = (0, _objectWithoutProperties2["default"])(props, _excluded);
  var _useStore = (0, _reactLoginPage.useStore)(),
    dispatch = _useStore.dispatch;
  if (!elmProps.children && visible) {
    elmProps.children = _email.IconEmail;
  }
  var key = keyname || name;
  var nameBase = name || rename || keyname;
  (0, _react.useEffect)(function () {
    return dispatch((0, _defineProperty2["default"])({}, "$$".concat(panel).concat(key), label));
  }, [label]);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactLoginPage.Input, (0, _objectSpread2["default"])((0, _objectSpread2["default"])({
    placeholder: "Your Email",
    spellCheck: false,
    visible: visible,
    index: 1
  }, elmProps), {}, {
    name: nameBase,
    keyname: "$$".concat(panel).concat(key)
  }));
});
Email.displayName = 'Login.Email';
